include/oneapi/dpl/pstl/utils_ranges.h

```diff
@@ -27,6 +27,35 @@ namespace oneapi
 {
 namespace dpl
 {
+
+namespace __internal
+{
+
+template <typename _R>
+auto
+get_value_type(int) -> typename ::std::decay_t<_R>::value_type;
+
+template <typename _R>
+auto
+get_value_type(long) ->
+    typename ::std::iterator_traits<::std::decay_t<decltype(::std::declval<_R&>().begin())>>::value_type;
+
+template <typename _R>
+auto
+get_value_type(...)
+{
+    //static_assert has to be evaluated to false always - it should depends on a template parameter
```

Comment:
This comment is pretty good. I am not sure if we should do it better. Something like "to have a correct behavior it should depend on template parameter.

And definitely fix the typo, please.

```suggestion
    //static_assert has to be evaluated to false always - it should depend on a template parameter
```

Reply:
the suggestion is not actual any more

Reply:
Something like "static_assert should always fail when this overload is chosen, so its condition must depend on the template parameter and evaluate to false". Not sure how to make it shorter without losing information.